### PR TITLE
Added python 3.9 to CI, tox and setup.py

### LINKED ISSUE #5620: Add Python 3.9 to CI
Python 3.9 was released on October 5, 2020. Add Python 3.9 to CI, tox configuration and update classifiers in setup.py

## Expected Result

What you expected.

## Actual Result

What happened instead.

## Reproduction Steps

Tests pass on Python 3.9.0

```
tox -e py39
GLOB sdist-make: /root/checked_repos/requests/setup.py
py39 create: /root/checked_repos/requests/.tox/py39
py39 inst: /root/checked_repos/requests/.tox/.tmp/package/1/requests-2.24.0.zip
py39 installed: certifi==2020.6.20,chardet==3.0.4,idna==2.10,requests @ file:///root/checked_repos/requests/.tox/.tmp/package/1/requests-2.24.0.zip,urllib3==1.25.10
py39 run-test-pre: PYTHONHASHSEED='2412316346'
py39 run-test: commands[0] | python setup.py test
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing requests.egg-info/PKG-INFO
writing dependency_links to requests.egg-info/dependency_links.txt
writing requirements to requests.egg-info/requires.txt
writing top-level names to requests.egg-info/top_level.txt
reading manifest file 'requests.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'requirements.txt'
writing manifest file 'requests.egg-info/SOURCES.txt'
running build_ext
========================================================================= test session starts ==========================================================================
platform linux -- Python 3.9.0, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
cachedir: .tox/py39/.pytest_cache
rootdir: /root/checked_repos/requests, configfile: pytest.ini
plugins: forked-1.3.0, httpbin-0.0.7, cov-2.10.1, mock-3.3.1, xdist-2.1.0
gw0 [555]g 0 items / 1 error                                                                                                                                           
................................................................................................................................................................ [ 28%]
...................................................................x............................................................................................ [ 57%]
.....................s......s................................................................................................................................... [ 86%]
............................................................ssssssssss.....                                                                                      [100%]
================================================================================ ERRORS ================================================================================
_________________________________________________________ ERROR collecting docs/_themes/flask_theme_support.py _________________________________________________________
docs/_themes/flask_theme_support.py:2: in <module>
    from pygments.style import Style
E   ModuleNotFoundError: No module named 'pygments'
======================================================================= short test summary info ========================================================================
ERROR docs/_themes/flask_theme_support.py - ModuleNotFoundError: No module named 'pygments'
=================================================== 542 passed, 12 skipped, 1 xfailed, 1 error in 132.95s (0:02:12) ====================================================
ERROR: InvocationError for command /root/checked_repos/requests/.tox/py39/bin/python setup.py test (exited with code 1)
_______________________________________________________________________________ summary ________________________________________________________________________________
ERROR:   py39: commands failed

```

### PR DESCRIPTION
**Pull request by hughesp8**
# Added Python 3.9 into CI, tox and setup.py

This PR is directly related to this issue #5620 

closes #5620 

**This PR is currently a draft**

I am creating the PR to run test suite/see how CI performs using python3.9 via Actions. I was unable to get a Python3.9 virtual environment working locally so this is my alternative to see if it works. (This is my first open-source contribution so if this approach is poor please go easy on me :) )

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,3 +24,5 @@ jobs:
     - stage: coverage
       python: '3.6'
       script: codecov
+    - stage: test
+      python: '3.9'
```
Comment (sigmavirus24) [inline .travis.yml]: This should move up before coverage otherwise 3.9's coverage will not be included in the data sent to `codecov`.

**Comment by sigmavirus24** [comment]
Looks like travis doesn't have 3.9 yet

**Comment by hughesp8** [comment]
> Looks like travis doesn't have 3.9 yet

I looked through the on-demand python versions for travis and they do have python-3.9-dev.tar.bz2. So the dev version. However I don't know how to target. Any ideas?

Here are the available versions https://docs.travis-ci.com/user/languages/python/#python-versions

File: .travis.yml [hunk 2]
```diff
@@ -21,6 +21,8 @@ jobs:
       dist: xenial
     - stage: test
       python: '3.8'
+    - stage: test
+      python: '3.9'
```
Comment (hugovk) [inline .travis.yml:25]: You can test against `3.9-dev` whilst we're waiting for them to add `3.9`:

```suggestion
      python: '3.9-dev'
```

btw this is the tracking issue to +1
* https://travis-ci.community/t/python-3-9-0-build/10091

Comment (sigmavirus24) [reply .travis.yml:25]: Often times dev and the final releases aren't quite the same. I think this happened recently with 3.8 and I'd rather not start testing against -dev and trying to fix those bugs

Comment (hugovk) [reply .travis.yml:25]: Sure.

On Travis, `-dev` is a daily build from the matching CPython branch.

On GitHub Actions, `-dev` is the last tagged release (e.g. alpha, beta, rc, final).

Comment (hughesp8) [reply .travis.yml:25]: Thanks for the feedback guys. In hindsight I should've realized I could just add ```-dev```.

> btw this is the tracking issue to +1

From the looks of this travis will have support for 3.9 very soon, I will update with ```-dev``` in the mean time if you guys are ok with it

**Comment by hugovk** [comment]
The Python 3.9 job failed trying to build greenlet from source:
```
[pipenv.exceptions.InstallError]: Collecting greenlet==0.4.15
[pipenv.exceptions.InstallError]:   Using cached greenlet-0.4.15.tar.gz (59 kB)
[pipenv.exceptions.InstallError]: Building wheels for collected packages: greenlet
[pipenv.exceptions.InstallError]:   Building wheel for greenlet (setup.py): started
[pipenv.exceptions.InstallError]:   Building wheel for greenlet (setup.py): finished with status 'error'
```

* https://travis-ci.com/github/psf/requests/jobs/416898376

The latest greenlet 0.4.17 has a (manylinux) wheel for Python 3.9:

* https://pypi.org/project/greenlet/0.4.17/#files

How about bumping to that version?

**Comment by hughesp8** [comment]
> The latest greenlet 0.4.17 has a (manylinux) wheel for Python 3.9:
@hugovk In order to bump the greenlet version I just changed it in the pipfile.lock directly. This caused all of the builds to fail. Is this not how it's supposed to be done?

**Comment by hugovk** [comment]
You'll have to look up how to bump a version in `Pipfile.lock`, it's generated and contains locked hashes.

**Comment by nateprewitt** [comment]
Hi @hughesp8,

We ended up having to do a larger refactor to remove Pipfile and Travis (#5652) for our upcoming 2.25.0 release. I'm afraid that's going to make this PR obsolete, but I wanted to say thanks for contributing! Hopefully, we'll have other opportunities for you to work on in the future.

**Pull request closed without merging**